owncloudApp/src/main/java/com/owncloud/android/presentation/ui/settings/fragments/SettingsFragment.kt

```diff
@@ -204,6 +221,112 @@ class SettingsFragment : PreferenceFragmentCompat() {
 
     }
 
+    private fun manageMoreSettings() {
+
+        prefMoreCategory = findPreference(PREFERENCE_MORE_CATEGORY)
+        prefHelp = findPreference(PREFERENCE_HELP)
+        prefSync = findPreference(PREFERENCE_SYNC_CALENDAR_CONTACTS)
+        prefRecommend = findPreference(PREFERENCE_RECOMMEND)
+        prefFeedback = findPreference(PREFERENCE_FEEDBACK)
+        prefPrivacyPolicy = findPreference(PREFERENCE_PRIVACY_POLICY)
+        prefImprint = findPreference(PREFERENCE_IMPRINT)
+        prefAboutApp = findPreference(PREFERENCE_ABOUT_APP)
+
+        // Help
+        if (settingsViewModel.isHelpEnabled()) {
+            prefHelp?.setOnPreferenceClickListener {
+                val helpUrl = settingsViewModel.getHelpUrl()
+                goToUrl(helpUrl)
+                true
+            }
+        } else {
+            prefMoreCategory?.removePreference(prefHelp)
+        }
+
+        // Sync contacts, calendars and tasks
+        if (settingsViewModel.isSyncEnabled()) {
+            prefSync?.setOnPreferenceClickListener {
+                val syncUrl = settingsViewModel.getSyncUrl()
+                goToUrl(syncUrl)
+                true
+            }
+        } else {
+            prefMoreCategory?.removePreference(prefSync)
+        }
+
+        // Recommend
+        if (settingsViewModel.isRecommendEnabled()) {
+            prefRecommend?.setOnPreferenceClickListener {
+                val intent = Intent(Intent.ACTION_SENDTO)
+                intent.data = Uri.parse(getString(R.string.mail_recommend))
+                intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+
+                val appName = getString(R.string.app_name)
+                val downloadUrl = getString(R.string.url_app_download)
+
+                val recommendSubject = String.format(getString(R.string.recommend_subject), appName)
+                val recommendText = String.format(getString(R.string.recommend_text), appName, downloadUrl)
+
+                intent.putExtra(Intent.EXTRA_SUBJECT, recommendSubject)
+                intent.putExtra(Intent.EXTRA_TEXT, recommendText)
+                startActivity(intent)
+                true
+            }
+        } else {
+            prefMoreCategory?.removePreference(prefRecommend)
+        }
+
+        // Feedback
+        if (settingsViewModel.isFeedbackEnabled()) {
+            prefFeedback?.setOnPreferenceClickListener {
+                val feedbackMail = getString(R.string.mail_feedback)
+                val feedback = "Android v" + BuildConfig.VERSION_NAME + " - " + getString(R.string.prefs_feedback)
+                val intent = Intent(Intent.ACTION_SENDTO)
```

Comment:
I can see that we build mail intents more than one time in the app (at least two times in this file). I think that we could unify them as you did when [mocking intents](https://github.com/owncloud/android/blob/settings%2Fandroidx/owncloudApp/src/androidTest/java/com/owncloud/android/utils/IntentsUtil.kt). You could create a new Fragment extension [here](https://github.com/owncloud/android/blob/settings%2Fandroidx/owncloudApp/src/main/java/com/owncloud/android/extensions/FragmentExt.kt) and keep learning Kotlin things 🍻

Reply:
Finally moved to Activity extension as we discussed, and new issue (https://github.com/owncloud/android/issues/3138) created to adapt the rest of the project